Shared/Extensions/URLExtensions.swift

```diff
@@ -320,7 +271,7 @@ extension URL {
     :returns: The public suffix for within the given hostname.
     */
   public var publicSuffix: String? {
-    return host.flatMap { publicSuffixFromHost($0, withAdditionalParts: 0) }
+    return host.flatMap { URLHelper.shared.publicSuffix(fromHost: $0, withAdditionalParts: 0) }
```

Comment:
FYI this API is being deprecated in this PR: https://github.com/brave/brave-ios/pull/5239/files

Reply:
> FYI this API is being deprecated in this PR: https://github.com/brave/brave-ios/pull/5239/files

I will need access to be able to extract the eTLD+1 from a string, not a URL. Do you know if I can do that with the new API?

Reply:
As far as I know, no, the Chromium API is based on URLs not strings. Is it not feasible to make a URL from the host?

cc @Brandon-T

Reply:
Ok, I reverted my URLExtension. For now I create a temporary url like `//youtube.com` to extract the ETLD and I will adjust it according to Brandon's feedback.